vars/buildDotNetCore.groovy

```diff
@@ -33,9 +33,11 @@ def call(Map config=[:]) {
         config['buildClosure']()
       }
 
-      stage('Build test image') {
-        build.buildTestImage(DOCKER_REGISTRY_CREDENTIALS_ID, DOCKER_REGISTRY, repoName, BUILD_NUMBER, tag)
-      }
+      if (fileExists('./docker-compose.test.yaml')) {
+        stage('Build test image') {
+          build.buildTestImage(DOCKER_REGISTRY_CREDENTIALS_ID, DOCKER_REGISTRY, repoName, BUILD_NUMBER, tag)
+        }
+      } 
```

Comment:
You can't wrap the provision resources step in this statement as we always want to provision the infrastructure for the PR deployment.  You could either two two if statements or move the build test image until after the provision resources step.